internal/api/impl/v1/view/service.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package view
+
+import (
+	"github.com/perses/perses/internal/api/interface/v1/view"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// A counter for the total number of views of a dashboard.
+var dashboardViewCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+	Name: "perses_dashboard_view_total",
```

Comment:
Why not defining a constant `perses` for the metric namespace ?
We could also use it when defining the http server here: https://github.com/perses/perses/blob/main/internal/api/core/core.go#L50

Reply:
Done. I've defined it as a namespace for these metrics - the actual flow of how we register metrics is a bit opaque, involving a magic import from perses/common, so we might want to rejig that flow in the future to make metrics registration a bit more explicit and consistent

Reply:
actually that shouldn't be that magic.

in `perses/common` we are providing default metrics (`_info` and metrics for the http server). Then it's simply possible to provide a prometheus metric registry to the `perses/common` app. If a metric registry is not provided, then the default one (provided by Prometheus) is used.

We are using a different metric registry in the e2e test: https://github.com/perses/perses/blob/main/internal/api/e2e/framework/server.go#L103

So today if we want to make metrics registration a bit more explicit and consistent we just need to define a new one and then register every metric in the `main` and then inject it in the `perses/common` app

Reply:
I'm currently patching a bit `perses/common` that contains a bug about how we are using the prometheus registry.
And then I will integrate it in Perses. Let see if you find it more coherent !